src/config.c

```diff
@@ -3184,6 +3191,7 @@ standardConfig static_configs[] = {
     createLongLongConfig("proto-max-bulk-len", NULL, DEBUG_CONFIG | MODIFIABLE_CONFIG, 1024*1024, LONG_MAX, server.proto_max_bulk_len, 512ll*1024*1024, MEMORY_CONFIG, NULL, NULL), /* Bulk request max size */
     createLongLongConfig("stream-node-max-entries", NULL, MODIFIABLE_CONFIG, 0, LLONG_MAX, server.stream_node_max_entries, 100, INTEGER_CONFIG, NULL, NULL),
     createLongLongConfig("repl-backlog-size", NULL, MODIFIABLE_CONFIG, 1, LLONG_MAX, server.repl_backlog_size, 1024*1024, MEMORY_CONFIG, NULL, updateReplBacklogSize), /* Default: 1mb */
+    createLongLongConfig("replica-full-sync-buffer-limit", NULL, MODIFIABLE_CONFIG, 0, LLONG_MAX, server.replica_full_sync_buffer_limit, 0, MEMORY_CONFIG, NULL, updateReplicaBufferLimit), /* Zero value means no limitations */
```

Comment:
this config isn't documented, but i wonder if we need / want it anyway, and if not, what should be the default value.
considering any default other than unlimited would be wrong, setting it too low can cause nearly every user to be forced to adjust it, setting it too high, is similar to setting it to unlimited, it'll not do enough to protect the system, and if users don't experience issues, they might not tune it.

today, replication buffers are excluded from eviction and OOM (because eviction can cause a feedback loop when issuing DELs), that's not the case with this buffer, however, replicas can't do eviction or OOM.

on the other hand, `client-output-buffer-limit` (for replicas) is by default set to 256MB, after which the client is disconnected.
maybe we can inherit the value from that config, which could already be tuned, instead of forcing people to set this one on upgrades?

Reply:
I agree with the above concerns, 
I don't like using `client-output-buffer-limit` because in most cases until the sync is done, this replica is worthless.
We may use `client-output-buffer-limit` in cases when the replica is currently serving clients.